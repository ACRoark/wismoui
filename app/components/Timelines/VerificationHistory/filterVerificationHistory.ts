import * as _ from 'lodash';
import IVerificationEventStatusUpdate from 'types/IVerificationEventStatusUpdate';

const filterVerificationHistory = (
  verificationEventStatusUpdates: IVerificationEventStatusUpdate[],
): IVerificationEventStatusUpdate[] => {
  const history: IVerificationEventStatusUpdate[] = [];

  verificationEventStatusUpdates.forEach((e: IVerificationEventStatusUpdate): void => {
    if (history.length === 0) {
      history.push(e);
    } else {
      const lastItem = _.last(history);

      if (lastItem.verificationCode === e.verificationCode) {
        if (
          lastItem.createdAtDate < e.createdAtDate ||
          (lastItem.createdAtDate === e.createdAtDate && lastItem.createdAtTime < e.createdAtTime)
        ) {
          _.remove(history, lastItem);
          history.push(e);
        }
      } else {
        history.push(e);
      }
    }
  });

  return history;
};

export default filterVerificationHistory;
