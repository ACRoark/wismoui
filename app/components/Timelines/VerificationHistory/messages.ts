import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.verification-history';

export default defineMessages({
  AD: {
    id: `${scope}.AD`,
  },
  AN: {
    id: `${scope}.AN`,
  },
  AP: {
    id: `${scope}.AP`,
  },
  CL: {
    id: `${scope}.CL`,
  },
  DR: {
    id: `${scope}.DR`,
  },
  ES: {
    id: `${scope}.ES`,
  },
  FV: {
    id: `${scope}.FV`,
  },
  FW: {
    id: `${scope}.FW`,
  },
  LD: {
    id: `${scope}.LD`,
  },
  PA: {
    id: `${scope}.PA`,
  },
  PR: {
    id: `${scope}.PR`,
  },
  RE: {
    id: `${scope}.RE`,
  },
  RF: {
    id: `${scope}.RF`,
  },
  SC: {
    id: `${scope}.SC`,
  },
});
