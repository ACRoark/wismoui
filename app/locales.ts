export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
