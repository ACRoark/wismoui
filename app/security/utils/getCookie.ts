const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
