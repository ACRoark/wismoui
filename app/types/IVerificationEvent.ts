import IVerificationEventBase from './IVerificationEventBase';
import IVerificationEventStatusUpdate from './IVerificationEventStatusUpdate';

interface IVerificationEvent extends IVerificationEventBase {
  verificationEventStatusUpdates: IVerificationEventStatusUpdate[];
}

export default IVerificationEvent;
