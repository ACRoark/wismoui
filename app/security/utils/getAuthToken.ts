import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
