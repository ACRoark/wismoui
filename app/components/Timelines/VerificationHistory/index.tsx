import Timeline from 'antd/lib/timeline';
import * as _ from 'lodash';
import React, { ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import IVerificationEvent from 'types/IVerificationEvent';
import IVerificationEventStatusUpdate from 'types/IVerificationEventStatusUpdate';

import filterVerificationHistory from './filterVerificationHistory';
import messages from './messages';

import '../index.less';

interface IVerificationHistoryProps {
  verificationEvents: IVerificationEvent[];
}

const VerificationHistory: React.FC<IVerificationHistoryProps> = (
  props: IVerificationHistoryProps,
): React.ReactElement => {
  const { verificationEvents } = props;
  const { verificationEventStatusUpdates } = _.last(verificationEvents);

  const history = filterVerificationHistory(verificationEventStatusUpdates);

  if (history.length) {
    return (
      <Timeline className="dte-wismo-timeline">
        {history.map(
          (update: IVerificationEventStatusUpdate): ReactElement => (
            <Timeline.Item
              color="gray"
              key={`${update.verificationCode}_${update.createdAtDate}_${update.createdAtTime}`}
            >
              <FormattedMessage
                {...messages[`${update.verificationCode}`]}
                values={{ date: <DateStamp value={update.createdAtDate} /> }}
              />
            </Timeline.Item>
          ),
        )}
      </Timeline>
    );
  }

  return <></>;
};

export default VerificationHistory;
