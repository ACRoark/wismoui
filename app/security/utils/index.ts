import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
