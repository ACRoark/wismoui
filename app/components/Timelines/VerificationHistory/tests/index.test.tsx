import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import IVerificationEvent from 'types/IVerificationEvent';

import VerificationHistory from '..';

const completedVerificationEvents: IVerificationEvent[] = [
  {
    verificationCategory: 'R975',
    verificationEventId: '123123',
    verificationEventStatusUpdates: [
      {
        verificationCode: 'AD',
        verificationEventStatus: 'ACTIVE',
        createdAtDate: '20200320',
        createdAtTime: '092215',
      },
      {
        verificationCode: 'AD',
        verificationEventStatus: 'COMPLETED',
        createdAtDate: '20200321',
        createdAtTime: '092215',
      },
      {
        verificationCode: 'PR',
        verificationEventStatus: 'ACTIVE',
        createdAtDate: '20200321',
        createdAtTime: '122215',
      },
      {
        verificationCode: 'PR',
        verificationEventStatus: 'COMPLETED',
        createdAtDate: '20200321',
        createdAtTime: '142215',
      },
    ],
  },
];

const incompleteVerificationEvents: IVerificationEvent[] = [
  {
    verificationCategory: 'R975',
    verificationEventId: '123123',
    verificationEventStatusUpdates: [
      {
        verificationCode: 'AD',
        verificationEventStatus: 'ACTIVE',
        createdAtDate: '20200320',
        createdAtTime: '092215',
      },
      {
        verificationCode: 'AD',
        verificationEventStatus: 'COMPLETED',
        createdAtDate: '20200321',
        createdAtTime: '092215',
      },
      {
        verificationCode: 'PR',
        verificationEventStatus: 'ACTIVE',
        createdAtDate: '20200321',
        createdAtTime: '122215',
      },
    ],
  },
];

const expectComponentToMatchSnapshot = (orderVerificationEvents: IVerificationEvent[], language: string): void => {
  const tree = createSnapshotWithIntl(<VerificationHistory verificationEvents={orderVerificationEvents} />, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrors = (language: string, orderVerificationEvents: IVerificationEvent[]): void => {
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<VerificationHistory verificationEvents={orderVerificationEvents}/>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('VerificationHistory', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when language is (${language}) and verification events has items in 'COMPLETED' status`, (): void =>
      shouldNotLogErrors(language, completedVerificationEvents));

    it(`should not log errors in console when language is (${language}) and verification events has no items in 'COMPLETED' status`, (): void =>
      shouldNotLogErrors(language, incompleteVerificationEvents));

    it(`should show verification events history when language is ${language} and verification events has items in 'COMPLETED' status`, (): void =>
      expectComponentToMatchSnapshot(completedVerificationEvents, language));

    it(`should show verification events history when language is ${language} and verification events has last item in 'ACTIVE' status`, (): void =>
      expectComponentToMatchSnapshot(incompleteVerificationEvents, language));
  });
});
