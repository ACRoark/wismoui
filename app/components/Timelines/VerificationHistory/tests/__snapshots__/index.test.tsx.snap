// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationHistory should show verification events history when language is en and verification events has items in 'COMPLETED' status 1`] = `
<ul
  className="ant-timeline dte-wismo-timeline"
>
  <li
    className="ant-timeline-item"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentation Required 
      03/21/20
    </div>
  </li>
  <li
    className="ant-timeline-item ant-timeline-item-last"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documents Received 
      03/21/20
    </div>
  </li>
</ul>
`;

exports[`VerificationHistory should show verification events history when language is en and verification events has last item in 'ACTIVE' status 1`] = `
<ul
  className="ant-timeline dte-wismo-timeline"
>
  <li
    className="ant-timeline-item"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentation Required 
      03/21/20
    </div>
  </li>
  <li
    className="ant-timeline-item ant-timeline-item-last"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documents Received 
      03/21/20
    </div>
  </li>
</ul>
`;

exports[`VerificationHistory should show verification events history when language is es and verification events has items in 'COMPLETED' status 1`] = `
<ul
  className="ant-timeline dte-wismo-timeline"
>
  <li
    className="ant-timeline-item"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentación Requerida 
      21/03/20
    </div>
  </li>
  <li
    className="ant-timeline-item ant-timeline-item-last"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentos Recibidos 
      21/03/20
    </div>
  </li>
</ul>
`;

exports[`VerificationHistory should show verification events history when language is es and verification events has last item in 'ACTIVE' status 1`] = `
<ul
  className="ant-timeline dte-wismo-timeline"
>
  <li
    className="ant-timeline-item"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentación Requerida 
      21/03/20
    </div>
  </li>
  <li
    className="ant-timeline-item ant-timeline-item-last"
  >
    <div
      className="ant-timeline-item-tail"
    />
    <div
      className="ant-timeline-item-head ant-timeline-item-head-gray"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    />
    <div
      className="ant-timeline-item-content"
    >
      Documentos Recibidos 
      21/03/20
    </div>
  </li>
</ul>
`;
